### ISSUE TITLE
Can't compile on current OSX+brew

### ISSUE THREAD

**Issue by trurebel** [PIVX-Project/PIVX#1840]
I've got a fully updated Mac and tried to use the "recipe" from your blob to compile with git clone. I have a vague recollection of successfully compiling last year, but now I'm getting some Boost errors I believe:

```
Making all in src
  CXX      libbitcoin_server_a-validationinterface.o
validationinterface.cpp:18:102: error: use of undeclared identifier '_1'
    g_signals.UpdatedBlockTip.connect(boost::bind(&CValidationInterface::UpdatedBlockTip, pwalletIn, _1));
                                                                                                     ^
validationinterface.cpp:19:102: error: use of undeclared identifier '_1'; did you mean 'boost::placeholders::_1'?
    g_signals.SyncTransaction.connect(boost::bind(&CValidationInterface::SyncTransaction, pwalletIn, _1, _2, _3));
                                                                                                     ^~
                                                                                                     boost::placeholders::_1
/usr/local/include/boost/bind/placeholders.hpp:46:38: note: 'boost::placeholders::_1' declared here
BOOST_STATIC_CONSTEXPR boost::arg<1> _1;
                                     ^
validationinterface.cpp:19:106: error: use of undeclared identifier '_2'; did you mean 'boost::placeholders::_2'?
    g_signals.SyncTransaction.connect(boost::bind(&CValidationInterface::SyncTransaction, pwalletIn, _1, _2, _3));
                                                                                                         ^~
                                                                                                         boost::placeholders::_2
/usr/local/include/boost/bind/placeholders.hpp:47:38: note: 'boost::placeholders::_2' declared here
BOOST_STATIC_CONSTEXPR boost::arg<2> _2;
                                     ^
validationinterface.cpp:19:110: error: use of undeclared identifier '_3'
    g_signals.SyncTransaction.connect(boost::bind(&CValidationInterface::SyncTransaction, pwalletIn, _1, _2, _3));
                                                                                                             ^
validationinterface.cpp:20:114: error: use of undeclared identifier '_1'
    g_signals.NotifyTransactionLock.connect(boost::bind(&CValidationInterface::NotifyTransactionLock, pwalletIn, _1));
                                                                                                                 ^
validationinterface.cpp:21:108: error: use of undeclared identifier '_1'
    g_signals.UpdatedTransaction.connect(boost::bind(&CValidationInterface::UpdatedTransaction, pwalletIn, _1));
                                                                                                           ^
validationinterface.cpp:22:96: error: use of undeclared identifier '_1'
    g_signals.SetBestChain.connect(boost::bind(&CValidationInterface::SetBestChain, pwalletIn, _1));
                                                                                               ^
validationinterface.cpp:23:90: error: use of undeclared identifier '_1'
    g_signals.Inventory.connect(boost::bind(&CValidationInterface::Inventory, pwalletIn, _1));
                                                                                         ^
validationinterface.cpp:24:105: error: use of undeclared identifier '_1'
    g_signals.Broadcast.connect(boost::bind(&CValidationInterface::ResendWalletTransactions, pwalletIn, _1));
                                                                                                        ^
validationinterface.cpp:25:96: error: use of undeclared identifier '_1'; did you mean 'boost::placeholders::_1'?
    g_signals.BlockChecked.connect(boost::bind(&CValidationInterface::BlockChecked, pwalletIn, _1, _2));
                                                                                               ^~
                                                                                               boost::placeholders::_1
/usr/local/include/boost/bind/placeholders.hpp:46:38: note: 'boost::placeholders::_1' declared here
BOOST_STATIC_CONSTEXPR boost::arg<1> _1;
                                     ^
validationinterface.cpp:25:100: error: use of undeclared identifier '_2'
    g_signals.BlockChecked.connect(boost::bind(&CValidationInterface::BlockChecked, pwalletIn, _1, _2));
                                                                                                   ^
validationinterface.cpp:27:99: error: use of undeclared identifier '_1'
    g_signals.BlockFound.connect(boost::bind(&CValidationInterface::ResetRequestCount, pwalletIn, _1));
                                                                                                  ^
validationinterface.cpp:31:102: error: use of undeclared identifier '_1'
    g_signals.BlockFound.disconnect(boost::bind(&CValidationInterface::ResetRequestCount, pwalletIn, _1));
                                                                                                     ^
validationinterface.cpp:33:99: error: use of undeclared identifier '_1'; did you mean 'boost::placeholders::_1'?
    g_signals.BlockChecked.disconnect(boost::bind(&CValidationInterface::BlockChecked, pwalletIn, _1, _2));
                                                                                                  ^~
                                                                                                  boost::placeholders::_1
/usr/local/include/boost/bind/placeholders.hpp:46:38: note: 'boost::placeholders::_1' declared here
BOOST_STATIC_CONSTEXPR boost::arg<1> _1;
                                     ^
validationinterface.cpp:33:103: error: use of undeclared identifier '_2'
    g_signals.BlockChecked.disconnect(boost::bind(&CValidationInterface::BlockChecked, pwalletIn, _1, _2));
                                                                                                      ^
validationinterface.cpp:34:108: error: use of undeclared identifier '_1'
    g_signals.Broadcast.disconnect(boost::bind(&CValidationInterface::ResendWalletTransactions, pwalletIn, _1));
                                                                                                           ^
validationinterface.cpp:35:93: error: use of undeclared identifier '_1'
    g_signals.Inventory.disconnect(boost::bind(&CValidationInterface::Inventory, pwalletIn, _1));
                                                                                            ^
validationinterface.cpp:36:99: error: use of undeclared identifier '_1'
    g_signals.SetBestChain.disconnect(boost::bind(&CValidationInterface::SetBestChain, pwalletIn, _1));
                                                                                                  ^
validationinterface.cpp:37:111: error: use of undeclared identifier '_1'
    g_signals.UpdatedTransaction.disconnect(boost::bind(&CValidationInterface::UpdatedTransaction, pwalletIn, _1));
                                                                                                              ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [libbitcoin_server_a-validationinterface.o] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

**Comment by furszy** [comment]
Hello, thanks for reporting. Try running it with #1899 on top.

**Comment by furszy** [comment]
Solved in #2082, closing.
